jobserver/models.py

```diff
@@ -108,44 +108,54 @@ def get_absolute_url(self):
         return reverse("job-detail", kwargs={"pk": self.pk})
 
     @property
-    def is_complete(self):
-        return self.status_code is None and self.completed_at
+    def is_completed(self):
+        if not self.is_finished:
+            return False
 
-    @property
-    def is_dependency_failed(self):
-        return self.status_code == 7
+        return self.status_code is None and self.completed_at
```

Comment:
Is there an argument for the implementations of these properties being via the queryset, e.g. `Job.objects.completed.filter(pk=self.pk).count() > 0` or similar?

Reply:
There's definitely an argument for it… which I've had many times before 😀 

I find the QuerySet version trickier to read myself and on a list page one would have to have knowledge of the property implementation to know that it will create O(N) queries so I try to avoid that style of lookup for something like this.